core/src/test/java/org/elasticsearch/search/SearchServiceTests.java

```diff
@@ -282,7 +282,10 @@ public FailOnRewriteQueryBuilder() {
 
         @Override
         protected QueryBuilder doRewrite(QueryRewriteContext queryShardContext) {
-            throw new IllegalStateException("Fail on rewrite phase");
+            if (queryShardContext.convertToShardContext() != null) {
```

Comment:
maybe rename queryShardContext to queryRewriteContext then to make it less confusing